File: aLi.py
import requests
import json
class ALi:
    sb = []

    @staticmethod
    def run_ali(refresh_token):
        update_access_token_url = "https://auth.aliyundrive.com/v2/account/token"
        sign_in_url = "https://member.aliyundrive.com/v1/activity/sign_in_list?_rx-s=mobile"
        reward_url = "https://member.aliyundrive.com/v1/activity/sign_in_reward?_rx-s=mobile"

        headers = {"Content-Type": "application/json"}
        refresh_token = refresh_token

        print("refresh_token: ", refresh_token)

        update_access_data = {
            "grant_type": "refresh_token",
            "refresh_token": refresh_token
        }

        update_access_data_entity = json.dumps(update_access_data)
        update_access_date_rep = None

        try:
            update_access_date_rep = requests.post(update_access_token_url, data=update_access_data_entity, headers=headers)
            update_access_token_rep = update_access_date_rep.json()
        except requests.exceptions.RequestException as e:
            ALi.sb.append("提示: TOKEN 失效，请更新！")
            print(ALi.sb)
            return

        ALi.sb.append("------------------获取个人信息成功--------------------\n")
        ALi.sb.append("用户名 ：" + str(update_access_token_rep["user_name"]) + "\n")

        headers["Authorization"] = update_access_token_rep["access_token"]

        sign_in_url_data = {"isReward": False}
        sign_in_url_data_entity = json.dumps(sign_in_url_data)
        sign_in_url_data_rep = None
        sign_in_url_list_rep = None

        try:
            sign_in_url_data_rep = requests.post(sign_in_url, data=sign_in_url_data_entity, headers=headers)
            sign_in_url_list_rep = sign_in_url_data_rep.json()
        except requests.exceptions.RequestException as e:
            ALi.sb.append("提示 : 签到列表获取失败 ！")
            print(ALi.sb)
            return

        sign_in_count = sign_in_url_list_rep["result"]["signInCount"]
        sign_in_logs = sign_in_url_list_rep["result"]["signInLogs"]
        ALi.sb.append("------------------获取签到列表信息成功------------------\n")
        ALi.sb.append("本月累计签到 ：" + str(sign_in_count) + "\n")

        sign_day_in_logs = sign_in_logs[int(sign_in_count) - 1]

        for i in range(len(sign_in_logs)):
            if sign_in_logs[i]["status"] == "normal" and not sign_in_logs[i]["isReward"]:
                day = sign_in_logs[i]["day"]
                reward_date = {"signInDay": day}
                reward_date_entity = json.dumps(reward_date)
                reward_date_rep = None
                reward_list_date_rep = None

                """
                try:
                    # reward_date_rep = requests.post(reward_url, data=reward_date_entity, headers=headers)
                    # reward_list_date_rep = reward_date_rep.json()
                    if reward_list_date_rep["success"]:
                        ALi.sb.append("------------------奖励领取成功--------------------\n")
                        ALi.sb.append("奖品名称 ：" + reward_list_date_rep["result"]["name"] + "\n")
                        ALi.sb.append("奖品名称 ：" + reward_list_date_rep["result"]["description"] + "\n")
                        ALi.sb.append("奖品名称 ：" + reward_list_date_rep["result"]["notice"] + "\n")
                        print(ALi.sb)
                        return
                except requests.exceptions.RequestException as e:
                    ALi.sb.append("提示 : 奖励领取失败 ！")
                    print(ALi.sb)
                    return
                """
        ALi.sb.append("------------------当天已领取奖励-------------------\n")
        reward_name = sign_day_in_logs.get("reward", {}).get("name")
        if reward_name is not None:
            ALi.sb.append("奖品名称 ：" + reward_name + "\n")
        else:
            ALi.sb.append("奖品名称 ：未知奖品\n")
        
        reward_description = sign_day_in_logs.get("reward", {}).get("description")
        if reward_description is not None:
            ALi.sb.append("奖品描述 ：" + reward_description + "\n")
        else:
            ALi.sb.append("奖品描述 ：未知描述\n")
        

    @staticmethod
    def send_email_content():
        res = "\n".join(ALi.sb)
        return res

File: src/main/java/org/dellevin/utils/aLi.java
package org.dellevin.utils;

import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.ResponseEntity;
import org.springframework.web.client.RestClientException;
import org.springframework.web.client.RestTemplate;

public class aLi {
    static final StringBuffer sb = new StringBuffer();
    public static void runALi(String refreshToken) {
        final String updateAccesssTokenURL = "https://auth.aliyundrive.com/v2/account/token";
        final String signinURL = "https://member.aliyundrive.com/v1/activity/sign_in_list?_rx-s=mobile";
        final String rewardURL = "https://member.aliyundrive.com/v1/activity/sign_in_reward?_rx-s=mobile";
        final RestTemplate re = new RestTemplate();
        final HttpHeaders headers = new HttpHeaders();
        String refresh_token =refreshToken;


        System.out.println("refresh_token:  "+refresh_token);
        headers.add("Content-Type", "application/json");

        JSONObject updateAccesssDate = new JSONObject();
        updateAccesssDate.put("grant_type", "refresh_token");
        updateAccesssDate.put("refresh_token", refresh_token);

        HttpEntity<String> updateAccesssDateEntity = new HttpEntity<String>(JSONObject.toJSONString(updateAccesssDate), headers);
        ResponseEntity<String> updateAccesssDateRep = null;
        try {
            updateAccesssDateRep = re.postForEntity(updateAccesssTokenURL, updateAccesssDateEntity, String.class);
        } catch (RestClientException e) {
            sb.append("提示 : TOKEN 失效，请更新 ！");
            System.out.println(sb);
            return;
        }
        JSONObject UpdateAccesssTokenRep = JSONObject.parseObject(updateAccesssDateRep.getBody());
        sb.append("------------------获取个人信息成功--------------------" + "\n");
        sb.append("用户名 ：" + UpdateAccesssTokenRep.getString("user_name") + "\n");

        headers.add("Authorization", UpdateAccesssTokenRep.getString("access_token"));
        JSONObject signinURLDate = new JSONObject();
        signinURLDate.put("isReward", false);
        HttpEntity<String> signinURLDateEntity = new HttpEntity<String>(JSONObject.toJSONString(signinURLDate), headers);
        ResponseEntity<String> signinURLDateRep = null;
        JSONObject signinURListRep = null;

        try {
            signinURLDateRep = re.postForEntity(signinURL, signinURLDateEntity, String.class);
            signinURListRep = JSONObject.parseObject(signinURLDateRep.getBody());
        } catch (RestClientException e) {
            sb.append("提示 : 签到列表获取失败 ！");
            System.out.println(sb);
            return;
        }
        String signInCount = signinURListRep.getJSONObject("result").getString("signInCount");
        JSONArray signInLogs = signinURListRep.getJSONObject("result").getJSONArray("signInLogs");
        sb.append("------------------获取签到列表信息成功------------------" + "\n");
        sb.append("本月累计签到 ：" + signInCount + "\n");
        //当天签到信息
        JSONObject signDayInLogs = signInLogs.getJSONObject(Integer.valueOf(signInCount) - 1);
        //获取未领取奖励列表
        
        for (int i = 0; i < signInLogs.size(); i++) {
            if (signInLogs.getJSONObject(i).getString("status").equals("normal") && signInLogs.getJSONObject(i).getString("isReward").equals("false")) {
                final String day = signInLogs.getJSONObject(i).getString("day");
                JSONObject rewardDate = new JSONObject();
                rewardDate.put("signInDay", day);
                HttpEntity<String> rewardDateEntity = new HttpEntity<String>(rewardDate.toJSONString(), headers);
                ResponseEntity<String> rewardDateRep = null;
                JSONObject rewardListDateRep = null;
                /*
                try {
                   // rewardDateRep = re.postForEntity(rewardURL, rewardDateEntity, String.class);
                  //  rewardListDateRep = JSONObject.parseObject(rewardDateRep.getBody());
                    if (rewardListDateRep.getString("success").equals("true")) {
                        sb.append("------------------奖励领取成功--------------------" + "\n");
                        sb.append("奖品名称 ：" + rewardListDateRep.getJSONObject("result").getString("name") + "\n");
                        sb.append("奖品名称 ：" + rewardListDateRep.getJSONObject("result").getString("description") + "\n");
                        sb.append("奖品名称 ：" + rewardListDateRep.getJSONObject("result").getString("notice") + "\n");
                        System.out.println(sb);
                        return;
                    }
                } catch (RestClientException e) {
                    sb.append("提示 : 奖励领取失败 ！");
                    System.out.println(sb);
                    return;
                }
                */
            }
        }
        sb.append("------------------当天已领取奖励-------------------" + "\n");
        sb.append("奖品名称 ：" + signDayInLogs.getJSONObject("reward").getString("name") + "\n");
        sb.append("奖品名称 ：" + signDayInLogs.getJSONObject("reward").getString("description") + "\n");
        System.out.println(sb);
    }

    public static String sendEmailContent() {
        String res = String.valueOf(sb);
        return res;
    }

}
